### PR TITLE
`dev` kind + new cert-manager

### PR DESCRIPTION
**Pull request by johnstonmatt**
- added `dev` kind
- added cert-manager improvements

### CODE REVIEW

**Comment by IamTamika** [comment]
Ready for review

File: src/cloud-init/microk8s/leader.yml.ts [hunk 1]
```diff
@@ -174,14 +183,28 @@ const getLeaderCloudInitYaml = (config: CNDIConfig) => {
...
+    `echo "Setting NFS as default storage class"`,
+    `while ! sudo microk8s kubectl patch storageclass nfs -p '{ "metadata": { "annotations": { "storageclass.kubernetes.io/is-default-class": "true" } } }'; do echo 'microk8s failed to install nfs, retrying in ${MICROK8S_INSTALL_RETRY_INTERVAL} seconds'; sleep ${MICROK8S_INSTALL_RETRY_INTERVAL}; done`,
+    `echo "NFS is now the default storage class"`,
+  ];
+
+  if (isDevCluster(config)) {
+    packages = ["apache2-utils"]; // no nfs-common on dev clusters
+    storageClassSetupCommands = [
+      `echo "Setting hostpath-storage as default storage class"`,
+      // TODO: fix the next 2 lines
```
Comment (johnstonmatt) [inline src/cloud-init/microk8s/leader.yml.ts]: dev mode should set hostpath-storage as the storage class

can you handle that for me @IamTamika ?

**Comment by johnstonmatt** [comment]
Changing a dev node's `memory` (and presumably any other property on the node) results in node replacement with a new IP. Is this avoidable?

**Comment by IamTamika** [comment]
It’s a bit unclear, I’ll check with a standard multi pass deploy
https://multipass.run/docs/modify-an-instance#heading--set-the-cpu-ram-or-disk-of-an-instance

This multi pass doc “suggests” that it wouldn’t be a full replace but this
second link  with the terraform multi pass provider  code looks like it
only has the option to replace.

https://github.com/larstobi/terraform-provider-multipass/blob/main/internal/provider/resource_instance.go

**Pull request merged**